### PR TITLE
feat: add lightbox lab

### LINKED ISSUE #455: chore: add another event lab to frontend cert
For this lab, we want to give them more practice working with events. 

Campers will have already complete the previous fundamentals modules(strings, variables, arrays, numbers, Math object, objects, etc) So you can see the sheet for more details to see the complete list of what is covered.
They will also be familiar with the following:

- addEventListener
- removeEventListener
- click events
- change events
- textContent
- innerHTML
- el.style

## Acceptance criteria

- [ ] create a new folder of the project in the `frontend-cert/js-projects` directory
- [ ] add your completed project in an `index.html`, `styles.css` and `script.js` files in that new folder
- [ ] create a markdown file for your proposed user stories. You should call it `user-stories.md` and place it in your new project folder

### PR DESCRIPTION
**Pull request by zairahira**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a [descriptive title](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request?id=prepare-a-good-pr-title) (**not** a vague title like `Update index.md`)

Closes #455

To-do:
- [ ] Move images to cdn

### CODE REVIEW

**Review by moT01 — COMMENTED** [review]
The prototype looks good. I think there's too many user stories here though, which makes it really restrictive. I'm not sure how much will be provided as boilerplate for the project. If the HTML is provided, we can remove all those user stories and focus on the JS ones. 

Here's some user stories I wrote up quick that you could maybe use:

1. You should have a `div` with a class of `gallery` within your `body`.
1. Within the `.gallery` element, you should have at least three image thumbnails, each with a class of `gallery-item`.
1. You should have a `div` with a class of `lightbox` within your `body`.
1. You should have a `span` with an `id` of `close` within your `.lightbox` element. You can use `&times;` as its text if you want.
1. You should have a `img` with an `id` of `lightbox-image` within your `.lightbox` element.
1. Your `.lightbox` element should cover the entire viewport, have a semi-transparent background, and center its content. Initially, its `display` property should be set to `none` to hide it.
1. When you click one of your `.gallery-item` elements, the `.lightbox` element’s `display` property should be set to `flex` to make it, and the two elements within it, visible.
1. When you click one of your `.gallery-item` elements, the `#lightbox-image` element’s `src` should be set to a full-size version of the image clicked.
1. When your `.lightbox` element is visible and you click the `.close` button or the `.lightbox` element, the `.lightbox` elements `display` should be set back to `none`.

I don't know if those cover everything, but it trims things down quite a bit. Feel free to use/modify them if you want @zairahira.

If the HTML isn't provided, I would not force users to use our images. I would supply them as an option.

**Comment by zairahira** [comment]
@moT01,
I have updated the user stories based on your suggestions. However, I would still ask the campers to use the provided images. Otherwise, it would be difficult to test which image is the thumbnail and which one is the full size.

@jdwilkin4,
Would we at any point provide the html in the boiler plate? If so, we can simplify user stories further.

**Comment by jdwilkin4** [comment]
Looking at the other labs in this module, it looks like the HTML is not provided to campers.
For consistency, we should have them add the relevant HTML.

So we can keep the updated user stories

File: frontend-cert/js-projects/lab-lightbox/user-stories.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+A lightbox displays a larger version of an image when clicked and shadows the rest of the page. This project will create a lightbox gallery that displays full-size images when a thumbnail is clicked. For each image, two versions are provided: a thumbnail and a full-size image. The full-size image is the same as the thumbnail, but without the `-thumbnail` suffix.
+
+1. You should have a `div` with a class of `gallery` within your `body`.
+2. Within the `.gallery` element, you should have at three image thumbnails, each with a class of `gallery-item`. You should use the following links for thumbnail images:
```
Comment (moT01) [inline frontend-cert/js-projects/lab-lightbox/user-stories.md:4]:
```suggestion
2. Within the `.gallery` element, you should have three image thumbnails, each with a class of `gallery-item`. You should use the following links for thumbnail images: 
```

I think if we're going to use these images, we can just make it so they only add the three?

File: frontend-cert/js-projects/lab-lightbox/user-stories.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+1. You should have a `div` with a class of `gallery` within your `body`.
+2. Within the `.gallery` element, you should have at three image thumbnails, each with a class of `gallery-item`. You should use the following links for thumbnail images: 
+<!-- would be replaced by cdn -->
+
+   - `./images/stonehenge-thumbnail.jpg`
+   - `./images/storm-thumbnail.jpg`
+   - `./images/trees-thumbnail.jpg`
+
+3. You should have a `div` with a class of `lightbox` within your `body`.
+4. You should have a `span` with an `id` of `close` within your `.lightbox` element. You can use `&times;` as its text if you want.??
```
Comment (moT01) [inline frontend-cert/js-projects/lab-lightbox/user-stories.md:12]:
```suggestion
4. You should have a `span` with an `id` of `close` within your `.lightbox` element. You can use `&times;` as its text if you want.
```

File: frontend-cert/js-projects/lab-lightbox/user-stories.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+5. You should have a `img` with an `id` of `lightbox-image` within your `.lightbox` element.
+6. Your `.lightbox` element should cover the entire viewport, have a semi-transparent background, and center its content. Initially, its `display` property should be set to `none` to hide it.
+7. When you click one of your `.gallery-item` elements, the `.lightbox` element’s `display` property should be set to `flex` to make it, and the two elements within it, visible.
+8. When you click one of your `.gallery-item` elements, the `#lightbox-image` element’s `src` should be set to a full-size version of the image clicked by removing `-thumbnail` from the image’s `src` attribute. The full-size images are located at the following links:
+<!-- would be replaced by cdn -->
+   - `./images/stonehenge.jpg`
+   - `./images/storm.jpg`
+   - `./images/trees.jpg`
+
+9.  When your `.lightbox` element is visible and you click the `.close` button or the `.lightbox` element, the `.lightbox` elements `display` should be set back to `none`.
```
Comment (moT01) [inline frontend-cert/js-projects/lab-lightbox/user-stories.md:22]:
```suggestion
9. When your `.lightbox` element is visible and you click the `.close` button or the `.lightbox` element, the `.lightbox` elements `display` should be set back to `none`.
```

**Review by moT01 — COMMENTED** [review]
Couple minor typos

**Review by moT01 — APPROVED** [review]
LGTM 🎉

**Pull request merged**